src/main/scala/is/hail/sparkextras/OrderedJoinDistinctIterator.scala

```diff
@@ -0,0 +1,105 @@
+package is.hail.sparkextras
+
+import is.hail.annotations.{JoinedRegionValue, RegionValue, UnsafeOrdering}
+
+abstract class OrderedJoinDistinctIterator(typ: OrderedRDD2Type, lrKOrd: UnsafeOrdering, leftIt: Iterator[RegionValue],
+  rightIt: Iterator[RegionValue]) extends Iterator[JoinedRegionValue] {
+  val jrv = JoinedRegionValue()
+  var lrv: RegionValue = _
+  var rrv: RegionValue = _
+
+  var present = false
+  var foundOne = false
```

Comment:
Something about the design of this feels wrong to me.  You never set this in the base case and that's quite confusing.

Reply:
How should I understand what `foundOne` is doing?